mrbgems/mruby-binding-core/src/binding-core.c

```diff
@@ -7,6 +7,18 @@
 #include <mruby/opcode.h>
 #include <mruby/debug.h>
 
+#define BINDING_UPPER_DEFAULT  20
+#define BINDING_UPPER_MINIMUM  10
+#define BINDING_UPPER_MAXIMUM 100
+
+#ifndef MRB_BINDING_UPPER_MAX
+# define BINDING_UPPER_MAX BINDING_UPPER_DEFAULT
+#else
+# define BINDING_UPPER_MAX ((MRB_BINDING_UPPER_MAX) > BINDING_UPPER_MAXIMUM ? BINDING_UPPER_MAXIMUM : \
+                            (MRB_BINDING_UPPER_MAX) < BINDING_UPPER_MINIMUM ? BINDING_UPPER_MINIMUM : \
+                            (MRB_BINDING_UPPER_MAX))
+#endif
```

Comment:
The user can define the depth of the proc with the `MRB_BINDING_UPPER_MAX` constant macro.

```console
% cat bindingtest.rb
bx = [binding]
50.times { |i|
  begin
    bx[-1].eval("var#{i} = #{i}")
    bx << bx[-1].dup
  rescue => e
    p [i, e]
    break
  end
}

% bin/mruby bindingtest.rb
[19, too many upper procs for local variables (mruby limitation; maximum is 20) (RuntimeError)]
```

Reply:
This part was wrong.
If `MRB_BINDING_UPPER_MAX` is defined, the `?:` operator is printed as is in the error message.
I will fix this.